core/update.php

```diff
@@ -531,46 +574,116 @@ function update_check_requirements($skip_warnings = FALSE) {
 if (update_access_allowed()) {
 
   include_once BACKDROP_ROOT . '/core/includes/install.inc';
+  include_once BACKDROP_ROOT . '/core/includes/backup.inc';
   include_once BACKDROP_ROOT . '/core/includes/batch.inc';
   backdrop_load_updates();
 
   update_fix_compatibility();
 
+  $op = isset($_REQUEST['op']) ? $_REQUEST['op'] : '';
+  $valid_token = isset($_GET['token']) && backdrop_valid_token($_GET['token'], 'update');
+
   // Check the update requirements for all modules. If there are warnings, but
   // no errors, skip reporting them if the user has provided a URL parameter
   // acknowledging the warnings and indicating a desire to continue anyway. See
   // backdrop_requirements_url().
-  $skip_warnings = !empty($_GET['continue']);
-  update_check_requirements($skip_warnings);
+  if (!$op || $op == 'info') {
+    $skip_warnings = !empty($_GET['continue']);
+    update_check_requirements($skip_warnings);
+  }
 
-  $op = isset($_REQUEST['op']) ? $_REQUEST['op'] : '';
   switch ($op) {
     // update.php ops.
+    case 'info':
+      $output = update_info_page();
+      break;
+
+    case 'check_updates':
+      $update_count = update_get_update_count();
+      if ($update_count === 0) {
+        backdrop_set_message(t('No pending updates.') . ' ' . t('All caches cleared.'));
+
+        // No updates to run, so caches won't get flushed later.  Clear them now.
+        backdrop_flush_all_caches();
+        install_goto('core/update.php?op=results');
+      }
+      else {
+        // Skip the backup and go to update selection if upgrading from Drupal 7.
+        $op = update_backup_enabled() ? 'backup' : 'selection';
+        $token = backdrop_get_token('update');
+        install_goto('core/update.php?op=' . $op . '&token=' . $token);
+      }
+      break;
+
+    case 'backup':
+      if ($valid_token) {
+        $output = update_backup_page();
+      }
+      else {
+        install_goto('core/update.php');
+      }
+      break;
+
+    case t('Create backup'):
```

Comment:
Hm... in this case the $op is a translatable string? Seems odd to me.

I know, "Apply pending updates" below does the same, but don't we have a key available, that doesn't rely on a translated string, which might - depending on language - contain all sorts of characters?

Reply:
> Or, asked the other way round - what's the benefit of using a translatable string as key?

Strange as it is, this is the way nearly all forms work in Backdrop because the label of a button is the submitted POST value. For forms that have multiple buttons, that is the only way to determine which button was clicked. That's not necessarily the case here (there's only a single button), but this is very common.

Reply:
Ah, many thanks for your explanations. Makes sense. Consider this question as answered / resolved. :+1:

Reply:
Blame the HTML spec for not having a separate label on `<input type="submit" value="Submit">` :stuck_out_tongue: